src/domain/community/contributor/Account/ContributorAccountView.tsx

```diff
@@ -217,12 +220,15 @@ export const ContributorAccountView = ({ accountHostName, account, loading }: Co
 
   const enableWingbackAccountCreation = !loading && !externalSubscriptionID;
 
-  // Temporarily we're ordering the priority in this way: Display usage/limit from FREE / PLUS / PREMIUM
-  const { limit: hostedSpaceLimit = 0, usage: hostedSpaceUsage = 0 } =
-    myAccountEntitlementDetails.find(entitlement => entitlement.type === LicenseEntitlementType.AccountSpaceFree) ??
-    myAccountEntitlementDetails.find(entitlement => entitlement.type === LicenseEntitlementType.AccountSpacePlus) ??
+  const { limit: spaceFreeLimit = 0, usage: spaceFreeUsage = 0 } =
+    myAccountEntitlementDetails.find(entitlement => entitlement.type === LicenseEntitlementType.AccountSpaceFree) ?? {};
+  const { limit: spacePlusLimit = 0, usage: spacePlusUsage = 0 } =
+    myAccountEntitlementDetails.find(entitlement => entitlement.type === LicenseEntitlementType.AccountSpacePlus) ?? {};
+  const { limit: spacePremiumLimit = 0, usage: spacePremiumUsage = 0 } =
     myAccountEntitlementDetails.find(entitlement => entitlement.type === LicenseEntitlementType.AccountSpacePremium) ??
     {};
```

Comment:
Maybe some iterations could be saved here with a `reduce` producing, but obviously that's not a big deal.